apptail/instance.go

```diff
@@ -79,10 +95,16 @@ func (instance *Instance) tailFile(name, filename string, stopCh chan bool) {
 		MaxLineSize: GetConfig().MaxRecordSize,
 		MustExist:   true,
 		Follow:      true,
-		Location:    &tail.SeekInfo{-limit, os.SEEK_END},
+		Location:    location,
 		ReOpen:      false,
 		Poll:        false,
 		RateLimiter: rateLimiter})
+
+	// IMPORTANT: this registration happens everytime app restarts
+	if shouldInitialize {
+		tracker.InitializeChildNode(instance.DockerId, filename, t.Location.Offset)
```

Comment:
Because the current offset you have says "negative 16 megabytes" or whatever; it's not the absolute position in the file at all.  (If you immediately restart `apptail` without reading anything first, it'll get serialized/deserialized and then try to see to negative 16MB from the start of the file, and fall over.)